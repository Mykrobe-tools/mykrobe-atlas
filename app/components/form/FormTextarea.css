.wrap {
  composes: wrap from 'styles/form.css';
}

.label {
  composes: label from 'styles/form.css';
}

.input {
  composes: input from 'styles/form.css';
}
