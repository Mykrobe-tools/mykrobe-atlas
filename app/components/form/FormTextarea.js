/* @flow */

import * as React from 'react';
import PropTypes from 'prop-types';

import FormLabel from './FormLabel';

import styles from './FormTextarea.css';

class FormTextarea extends React.Component {
  render() {
    const {
      name,
      onChange,
      placeholder,
      resize,
      rows,
      title,
      value,
    } = this.props;
    return (
      <div className={styles.wrap}>
        <div className={styles.label}>
          <FormLabel htmlFor={name} label={title} />
        </div>
        <textarea
          className={styles.input}
          id={name}
          name={name}
          rows={rows}
          style={resize ? null : { resize: 'none' }}
          value={value}
          placeholder={placeholder}
          onChange={event => onChange(event)}
        />
      </div>
    );
  }
}

FormTextarea.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  resize: PropTypes.string,
  rows: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
};

export default FormTextarea;
