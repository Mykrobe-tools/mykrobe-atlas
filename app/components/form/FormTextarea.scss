.wrap {
  composes: wrap from 'styles/form.scss';
}

.label {
  composes: label from 'styles/form.scss';
}

.input {
  composes: input from 'styles/form.scss';
}
