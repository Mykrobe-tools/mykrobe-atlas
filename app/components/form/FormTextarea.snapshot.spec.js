import React from 'react'
import renderer from 'react-test-renderer'
import FormTextarea from './FormTextarea'

describe('FormTextarea component snapshot', () => {
  const name = 'loremIpsum'
  const title = 'Lorem Ipsum'
  const rows = '10'
  const placeholder = 'Lorem Ipsum Dolor Sit Amet'
  const value = 'Dolor sit amet'
  const onChange = jest.fn()

  it('renders correctly', () => {
    const component = renderer.create(
      <FormTextarea
        name={name}
        title={title}
        value={value}
        placeholder={placeholder}
        rows={rows}
        onChange={onChange} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
