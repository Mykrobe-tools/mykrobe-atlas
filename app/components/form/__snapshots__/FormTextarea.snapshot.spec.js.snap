exports[`FormTextarea component snapshot renders correctly 1`] = `
<div
  className="wrap">
  <div
    className="label">
    <label
      className="label"
      htmlFor="loremIpsum">
      Lorem Ipsum
    </label>
  </div>
  <textarea
    className="input"
    id="loremIpsum"
    name="loremIpsum"
    onChange={[Function]}
    placeholder="Lorem Ipsum Dolor Sit Amet"
    rows="10"
    style={
      Object {
        "resize": "none",
      }
    }
    value="Dolor sit amet" />
</div>
`;
